### PR TITLE
Add fluentd as a logger option

### PR DESCRIPTION
**Pull request by papey**
### What does this PR do?

This PR adds fluentd support to Traefik logger. (discussed on Discord for Traefik Hackaethon)

### Motivation

This PR is inspired by https://github.com/traefik/traefik/issues/2151

### More

- [ ] Added/updated tests
- [ ] Added/updated documentation

### Additional Notes

IMO, the main thing we need to discuss is how configuration is presented to end-user.

If needed I can work on multiple output support as requested in #2151

### CODE REVIEW

**Comment by papey** [comment]
In order to not overload the CI with small iterations, a draft branch is available here : https://github.com/bearstech/traefik/tree/add-fluentd-draft

The discussion about the feature is also moved to #2151

File: pkg/log/fluentd.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package log
+
+import (
+	"github.com/evalphobia/logrus_fluent"
+	"github.com/sirupsen/logrus"
+)
+
+// NewFluentHook return a fluentd hook for logrus loggin
+func NewFluentHook(level logrus.Level) (*logrus_fluent.FluentHook, error) {
+	// add fluentd hooks if specified in configuration
```
Comment (ad-m) [inline pkg/log/fluentd.go:10]: Either I do not see Traefik in terms of (I am familiar with whole traefik codebase), or the comment "if specified in configuration" seems to me also (yet) incorrect.

Comment (papey) [reply pkg/log/fluentd.go:10]: This is WIP, and it was a comment left here as a reminder. For now, everything is done in https://github.com/bearstech/traefik/tree/add-fluentd-draft in order to discuss and find the best implementation for this feature.

Comment (ldez) [reply pkg/log/fluentd.go:10]: @papey to avoid this kind of issue, could close the PR for now?

Comment (papey) [reply pkg/log/fluentd.go:10]: Sure.

**Comment by papey** [comment]
See #2151

File: pkg/log/fluentd.go (hunk 1 above)
Comment (ldez) [reply pkg/log/fluentd.go:10]: thank you

**Pull request closed without merging**